src/View.cc

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "ignition/gazebo/detail/View.hh"
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace detail
+{
+//////////////////////////////////////////////////
+View::View(std::set<ComponentTypeId> _compIds)
+{
+  this->componentTypes = _compIds;
+}
+
+//////////////////////////////////////////////////
+bool View::HasCachedComponentData(const Entity _entity) const
+{
+  auto cachedComps =
+    this->validData.find(_entity) != this->validData.end() ||
+    this->invalidData.find(_entity) != this->invalidData.end();
+  auto cachedConstComps =
+    this->validConstData.find(_entity) != this->validConstData.end() ||
+    this->invalidConstData.find(_entity) != this->invalidConstData.end();
+
+  if (cachedComps && !cachedConstComps)
+  {
+    ignwarn << "Non-const component data is cached for entity " << _entity
+      << ", but const component data is not cached." << std::endl;
+  }
+  else if (cachedConstComps && !cachedComps)
+  {
+    ignwarn << "Const component data is cached for entity " << _entity
+      << ", but non-const component data is not cached." << std::endl;
+  }
+
+  return cachedComps && cachedConstComps;
+}
+
+//////////////////////////////////////////////////
+bool View::RemoveEntity(const Entity _entity)
+{
+  this->invalidData.erase(_entity);
+  this->invalidConstData.erase(_entity);
+  this->missingCompTracker.erase(_entity);
+
+  if (!this->HasEntity(_entity) && !this->IsEntityMarkedForAddition(_entity))
+    return false;
+
+  this->entities.erase(_entity);
+  this->newEntities.erase(_entity);
+  this->toRemoveEntities.erase(_entity);
+  this->toAddEntities.erase(_entity);
+  this->validData.erase(_entity);
+  this->validConstData.erase(_entity);
+
+  return true;
+}
+//////////////////////////////////////////////////
```

Comment:
```suggestion
}

//////////////////////////////////////////////////
```
nit: add newline to separate the two methods

Reply:
Done in https://github.com/ignitionrobotics/ign-gazebo/pull/1317/commits/16052878db2f1516a32ef6e4e5882d9eacba2005